File: 3332-minimum-operations-to-exceed-threshold-value-ii/minimum-operations-to-exceed-threshold-value-ii.py
class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        import heapq

        heap = []
        for value in nums:
            heapq.heappush(heap, value)
        
        operations_count = 0
        while len(heap) > 1:
            x = heapq.heappop(heap)
            if x >= k:
                break
            y = heapq.heappop(heap)
            heapq.heappush(heap, min(x, y) * 2 + max(x, y))
            operations_count += 1
        
        return operations_count

File: 3332-minimum-operations-to-exceed-threshold-value-ii/minimum-operations-to-exceed-threshold-value-ii.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int k) {
        priority_queue<long long, vector<long long>, greater<long long>> heap;
        for(const long long value : nums) {
            heap.push(value);
        }
        int operations_count = 0;
        while(heap.size() > 1) {
            long long x = heap.top();
            heap.pop();
            if (x >= k)break;
            long long y = heap.top();
            heap.pop();
            heap.push(min(x, y) * 2LL + max(x, y));
            operations_count += 1;
        }
        return operations_count;
    }
};